src/matrixeventprocessor.ts

```diff
@@ -418,13 +442,6 @@ export class MatrixEventProcessor {
             }
             // Let it fall through.
         }
-        if (!profile) {
```

Comment:
we assume here now that the `profile` exists, but it can still not exist due to the code up there? why remove this?

Reply:
There isn't any need to attempt to fetch the profile twice? If we failed the first time, we just use the mxid and continue.